### DISCUSSION TITLE
Dev: Fragen zur Entwicklung, Webhook

### DISCUSSION THREAD

**Discussion by RaHehl** [jomjol/AI-on-the-edge-device#3164] [General] [answered]
Hallo,

ich habe mal einen versuch begonnen meinen Feature Request #3148 selbst umzusetzen (Disclaimer: ich hab keine Ahnung von C++):

[PR #3163 ](https://github.com/jomjol/AI-on-the-edge-device/pull/3163/files)

Die Firmware baut schonmal, lässt sich Flashen, und scheint grundsätzlich lauffähig zu sein.

Möglicherweise klappt auch der request schon (auch wenn da noch paar daten fehlen) aktuell erstmal auf die schnelle gegen den hoppscotch echo server https://echo.hoppscotch.io

"[0d00h01m14s] 2024-08-06T23:00:29	<DBG>	[FLOWCTRL] Status: Sending Webhook (23:00:29)" 

ohne error danach.

Allerdings sehe ich in der WebGui unter Configuration keinen Webhook Eintrag, was könnte ich da am HTML falsch gemacht haben?

**Comment by SybexX** [comment, marked as answer]
```
function UpdateAfterCategoryCheck() {
    ReadParameterAll();
    category["Analog"]["enabled"] = document.getElementById("Category_Analog_enabled").checked;
    category["Digits"]["enabled"] = document.getElementById("Category_Digits_enabled").checked;
    category["MQTT"]["enabled"] = document.getElementById("Category_MQTT_enabled").checked;
    category["InfluxDB"]["enabled"] = document.getElementById("Category_InfluxDB_enabled").checked;
    category["InfluxDBv2"]["enabled"] = document.getElementById("Category_InfluxDBv2_enabled").checked;
    category["GPIO"]["enabled"] = document.getElementById("Category_GPIO_enabled").checked;
    category["Webhook"]["enabled"] = document.getElementById("Category_Webhook_enabled").checked;   <<<<<<< dies fehlt noch

    UpdateInput();
    var sel = document.getElementById("Numbers_value1");
    UpdateInputIndividual(sel);
}
```

**Comment by RaHehl** [comment]
danke, hab grade gesehen das ich wohl auch die falsche html datei angepasst habe sd-card/html/edit_config_template.html anstelle sd-card/html/edit_config_param.html

**Reply by SybexX** [reply]
die edit_config_param.html wird eigentlich aus der edit_config_template.html erstellt, wegen den $TOOLTIPs
dafür ist AI-on-the-edge-device-webhook\tools\parameter-tooltip-generator\generate-param-doc-tooltips.py oder AI-on-the-edge-device-webhook\param-docs\generate-template-param-doc-pages.py da

**Comment by RaHehl** [comment]
Bisher funktioniert das schonmal wunderbar:

![image](https://github.com/user-attachments/assets/20c4cceb-fc8a-463e-b4e9-b9966756715d)

Der Request mit dem aktuellen wert kommt inklusive API Key im Header beim aktuellen php testscript auf dem webspace an und wird weggeschrieben.

Mal schauen wie ich das noch mit dem restlichen zeug angereicht und in json gepresst bekomme, dann hab ich im Grunde schon was ich wollte

**Comment by RaHehl** [comment]
#3163 funktioniert soweit, ich mach mal hier wieder zu
